R/Spectra.R

```diff
@@ -499,11 +500,27 @@ NULL
 #'   access and it is possible to *revert* the operation with the `reset`
 #'   function (see description of `reset` above).
 #'
+#' - `combinePeaks`: combines mass peaks within each spectrum with a difference
+#'   in their m/z values that is smaller than the maximal acceptable difference
+#'   defined by `ppm` and `tolerance`. Parameters `intensityFun` and `mzFun`
+#'   allow to define functions to aggregate the intensity and m/z values for
+#'   each such group of peaks. With `weighted = TRUE` (the default), the m/z
+#'   value of the combined peak is calculated using an intensity-weighted mean
+#'   and parameter `mzFun` is ignored. The [MsCoreUtils::group()] function is
+#'   used for the grouping of mass peaks. Parameter `msLevel.` allows to
+#'   define selected MS levels for which peaks should be combined. This
+#'   function returns a `Spectra` with the same number of spectra than the
+#'   input object, but with possibly combined peaks within each spectrum.
+#'   Additional peak variables (other than `"mz"` and `"intensity"`) are
+#'   dropped (i.e. their values are replaced with `NA`) for combined peaks. See
```

Comment:
Should this not be:

Additional peak variables (other than `"mz"` and `"intensity"`) are dropped (i.e. their values are replaced with `NA`) for combined peaks unless they are constant across the combined peaks.

Reply:
The current implementation just drops them without checking if they are constant (for peaks that are combined).

Reply:
changed it [now](https://github.com/rformassspectrometry/Spectra/pull/296/commits/74ffe66f78578a35313e10cc034eeb0d095a4123) to match that: values for peak variables are only replaced with `NA` if, across the combined peaks, they are not constant.